src/server.rs

```diff
@@ -1,89 +1,138 @@
+use std::{ffi::c_void, ptr};
+
+use open62541_sys::{UA_Server_runUntilInterrupt, __UA_Server_addNode, __UA_Server_write};
+
 use crate::{ua, DataType, Error, ObjectNode, Result, VariableNode};
 
+/// OPC UA server.
+///
+/// This represents an OPC UA server. Nodes can be added through the several methods below, and then
+/// the server can be started with [`run()`](Self::run).
 pub struct Server(ua::Server);
 
 impl Server {
-    /// Creates a new OPC UA server.
-    ///
-    /// # Errors
-    ///
-    /// When the OPC-UA server fails to initialize.
-    ///
-    pub fn new() -> Result<Self> {
-        Ok(Self(ua::Server::new()))
+    /// Creates server.
+    #[must_use]
+    pub fn new() -> Self {
+        Self(ua::Server::new())
     }
 
-    ///
-    /// Adds a new variable to the address space
+    /// Adds object node to address space.
     ///
     /// # Errors
     ///
-    /// When a error happens adding the node
-    ///
-    pub fn add_variable_node(&mut self, node: VariableNode) -> Result<()> {
-        let status_code = self.0.add_variable_node(
-            node.requested_new_node_id,
-            node.parent_node_id,
-            node.reference_type_id,
-            node.browse_name,
-            node.type_definition,
-            node.attributes,
-        );
+    /// This fails when the node cannot be added.
+    pub fn add_object_node(&mut self, node: ObjectNode) -> Result<()> {
+        let status_code = ua::StatusCode::new(unsafe {
```

Comment:
Wouldn't it be desirable to restrict the unsafe parts to `ua::Server`? Unfortunately, I don't know how to achieve this :sweat_smile: So let's keep it here for now.

Maybe add a TODO comment to explain this decision?